lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const puppeteer = require('puppeteer');
+const fs = require('fs');
+
+/** @typedef {{result?: {value?: string, objectId?: number}, exceptionDetails?: object}} ProtocolResponse */
+
+/**
+ * https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/test_runner/TestRunner.js;l=170;drc=f59e6de269f4f50bca824f8ca678d5906c7d3dc8
+ * @param {Record<string, function>} receiver
+ * @param {string} methodName
+ * @param {function} override
+ */
+function addSniffer(receiver, methodName, override) {
+  const original = receiver[methodName];
+  if (typeof original !== 'function') {
+    throw new Error('Cannot find method to override: ' + methodName);
+  }
+
+  receiver[methodName] = function() {
+    let result;
+    try {
+      // eslint-disable-next-line prefer-rest-params
+      result = original.apply(this, arguments);
+    } finally {
+      receiver[methodName] = original;
+    }
+    // In case of exception the override won't be called.
+    try {
+      // eslint-disable-next-line prefer-rest-params
+      Array.prototype.push.call(arguments, result);
+      // eslint-disable-next-line prefer-rest-params
+      override.apply(this, arguments);
+    } catch (e) {
+      throw new Error('Exception in overriden method \'' + methodName + '\': ' + e);
+    }
+    return result;
+  };
+}
+
+const sniffLhr = `
+new Promise(resolve => {
+  (${addSniffer.toString()})(
+    UI.panels.lighthouse.__proto__,
+    '_buildReportUI',
+    (lhr, artifacts) => resolve(JSON.stringify(lhr))
+  );
+});
+`;
+
+const startLighthouse = `
+(() => {
+  UI.ViewManager.instance().showView('lighthouse');
+  UI.panels.lighthouse.contentElement.querySelector('button').click();
+})()
+`;
+
+async function run() {
+  const browser = await puppeteer.launch({
+    executablePath: process.env.CHROME_PATH,
+    devtools: true,
+  });
+  const page = await browser.newPage();
+
+  // Cut off JS for initial page navigation.
+  // This step is unnecessary for every page I tried except https://cnn.com.
```

Comment:
What happens when you allow JS? This is rather surprising to me.

Reply:
LH never starts because cnn takes too long to load. I believe the page's scripts are hogging the main thread so much that `startLighthouse` never runs successfully.

Every other page I tested didn't need this step, but disabling JS still decreased the startup time for those pages.

Reply:
Yikes. I don't love doing this because if *we* can't run Lighthouse because a page is so busy, it might be indicative of our users not being able to run Lighthouse because a page is too busy. 

Finding the "Lighthouse doesn't work in scenario X" is actually the point of creating a DevTools repro script, so perhaps we've already found our first bug :)

Reply:
Good point, I took another look and it turns out I was overthinking the problem. The script just blocks on `await page.goto(...)` and never gets to the part where it tries to start LH.

Instead of disabling javascript to get the page to load faster, I just took out the `await` on `page.goto(...)`.

Reply:
Update: Trying to start LH immediately after navigation starts sometimes causes DT to hang on "Lighthouse is warming up..." forever.

Changed this to wait for `DOMContentLoaded` before trying to start LH.